QA/PoroDevTest/src/main/java/common/ui_setup/pom_setup/PoroDevPom/LoginPage.java

```diff
@@ -1,55 +1,76 @@
 package common.ui_setup.pom_setup.PoroDevPom;
 
+import common.ui_setup.DriverSetup;
 import common.ui_setup.SetupConstants;
 import common.ui_setup.pom_setup.BasePage;
 import org.openqa.selenium.WebDriver;
 import org.openqa.selenium.WebElement;
 import org.openqa.selenium.support.FindBy;
-import java.io.IOException;
+import org.testng.Assert;
+
+import java.util.List;
 
 public class LoginPage extends BasePage {
 
     public LoginPage(WebDriver driver) {
         super(driver);
-        driver.get(SetupConstants.BASE_URL);
+    }
+
+    public LoginPage(WebDriver driver, String url) {
+        super(driver);
+        DriverSetup.navigateToUrl(SetupConstants.BASE_URL, driver);
     }
 
     //element section
-    @FindBy(id = "basic_email")
-    WebElement emailField;
-    @FindBy(id = "basic_password")
-    WebElement passField;
+    @FindBy(name = "email")
+    WebElement we_emailInputField;
+    @FindBy(name = "password")
+    WebElement we_passInputField;
     @FindBy(className = "ant-btn-primary")
-    WebElement loginButton;
+    WebElement we_loginButton;
     //ASSERTION ELEMENTS
-    @FindBy(xpath = "//div[contains(text(),'Please input your username')]")
-    public WebElement emailErrorMsg;
-    @FindBy(xpath = "//div[contains(text(),'Please input your password')]")
-    public WebElement passErrorMsg;
-    @FindBy(xpath = "//span[contains(text(),'Request failed with status code 401')]")
-    public WebElement errorAfterBadCredentials;
-    @FindBy(className = "hvVmBA")
-    public WebElement userFirstNameAfterLogin;
+    @FindBy(xpath = "//body/div[@id='root']/div[1]/div[1]/form[1]/div[1]/span[2]")
+    WebElement we_email_requiredMsg;
+    @FindBy(xpath = "//body/div[@id='root']/div[1]/div[1]/form[1]/div[2]/span[3]")
+    WebElement we_pass_requiredMsg;
+    @FindBy(xpath = "//span[contains(text(),'Email is invalid')]")
+    WebElement we_email_invalidMsg;
+    @FindBy(xpath = "//span[contains(text(),'Wrong password')]")
```

Comment:
Locators which include text fragments should be avoided. Is there any better solution for this?

Reply:
I implemented assertion for the error messages in the other way, so I don't actually use these elements but I left them just in case. I will need suggestion about how to catch some of the elements because all there is is class name which is same for every error message, so I put them all in a List and implemented method to assert current one.